### DISCUSSION TITLE
Appropriate validation of users in Vercel Analytics

### DISCUSSION THREAD

**Discussion by SohamP2812** [vercel/community#1021] [Help] [answered]
Hey Vercel Team!

Just wanted to quickly mention that the new Vercel analytics integration in Next.JS is awesome! Gonna love using this for my projects.

I had a quick question about some stats I was looking at in my Vercel analytics dashboard for one of my projects. It seems to be counting visits from puppeteers and headless crawlers as page visits - and it is slightly distorting the analytics I am receiving (especially with respect to Google Analytics). 

Here is a screenshot of what seems strange to me:

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/69543944/199400895-ef8684f5-ecd3-452d-b351-a35412462f3a.png">

Wondering if this is intended or not - or perhaps indicative of something wrong with my web app?

**Comment by amyegan** [comment]
Glad to hear you like the new Analytics features @SohamP2812! I don't know exactly what is being counted, so I'm sharing this with the experts on the Analytics team to get their input.

**Comment by dglsparsons** [comment, marked as answer]
Hi @SohamP2812, we currently detect bot traffic based on their user-agent, and bot traffic is excluded automatically.  

Are you running Puppeteer/Playwright/Selenium tests against your non-production deployments by any chance? I don't believe automated testing counts as bot traffic, and we currently don't separate out production/preview traffic, so it could be this?

**Reply by SohamP2812** [reply]
Huh that is so strange! No I am not running any tests or using puppeteer anywhere. Is there any chance we could look at where the headless traffic is even coming from? I don't see this on Google Analytics either

**Reply by SohamP2812** [reply]
<img width="1227" alt="image" src="https://user-images.githubusercontent.com/69543944/199528210-535b969a-0535-4534-9346-2698e795aed6.png">
This was just in last 24 hours lol

**Reply by dglsparsons** [reply]
Interesting! It's possible we run some tools against deployments internally. I'll have a dig into what this could be and report back to you.

**Reply by SohamP2812** [reply]
Would really appreciate that! Let me know if you need info on what project/account I was using.

**Reply by dglsparsons** [reply]
Hey, i've had a look into this and it doesn't seem like it's anything we do internally as part of the Vercel platform - those requests are always counted as bot traffic and disregarded for the sake of analytics. 

Are you able to share a deployment URL or project_id so I can investigate the traffic for your site some more? 

Thanks :)

**Reply by SohamP2812** [reply]
Ohh that's interesting. Yeah for sure!
It's live at http://www.peakgaming.ca/ and https://peak-sohamp2812.vercel.app/

**Reply by dglsparsons** [reply]
Hey, at a look there's nothing overly suspicious about the traffic you're receiving - it just looks like someone/something is scraping your site. 

We're intending to make a change to filter out headless traffic, as it should be counted as bot traffic (which we exclude by default). Hopefully that will make your dashboard appear a bit more normal :)

**Reply by SohamP2812** [reply]
Ahh ok thank you for looking into it - good to hear it's nothing super suspicious. 

Great to hear that change will be coming, appreciate the investigation and response on your part! 

Thanks a ton again, looking forward for the filter change :)

**Reply by dglsparsons** [reply]
This should now be in place - so no more headless traffic! 🎊

**Reply by SohamP2812** [reply]
Amazing! Just checked it out and it looks great. Thanks a whole ton!
